userspace/engine/rule_loader_collector.cpp

```diff
@@ -48,8 +48,14 @@ static inline void define_info(indexed_vector<T>& infos, T& info, uint32_t id)
 	}
 }
 
-template <typename T>
-static inline void append_info(T* prev, T& info, uint32_t id)
+template <typename T, typename U>
+static inline void append_info(T* prev, U& info, uint32_t id)
+{
+	prev->visibility = id;
+}
+
+template <typename T, typename U>
+static inline void replace_info(T* prev, U& info, uint32_t id)
```

Comment:
Note for the reviewer: the addition of `replace_info` could be cargo cult programming, I am not sure if we need to do more here or if this is useful.

Reply:
After speaking with @jasondellaluce , this seems the right approach.